### PR TITLE
[apiRefactor] addComponent/removeComponent renaming

### PR DESCRIPTION
**Pull request by rcchen**
On `table.ts`:
* `addComponent(row, col, component)` is now `addComponent(component, row, col)`
* `_removeComponent(component)` is now `removeComponent(component)`

On `componentContainer.ts`:
* `addComponent(component)` is now `add(component)`
* `removeComponent(component)` is now `remove(component)`

On `component.ts` and descendants:
* `remove()` is now `destroy()`

### CODE REVIEW

**Comment by jtlan** [comment]
`_removeComponent` is defined on `ComponentContainer`, so we should do the rename at that level.

File: src/components/componentContainer.ts [hunk 1]
```diff
@@ -75,10 +96,5 @@ module Plottable {
       this.components().slice().forEach((c: Component) => c.detach());
       return this;
     }
-
-    public remove() {
```
Comment (jtlan) [inline src/components/componentContainer.ts]: Whoa there, we need this functionality.

**Comment by jtlan** [comment]
Rename `Component.remove()` to `Component.destroy()`, and then put the functionality back into `Table`.

File: src/components/componentContainer.ts (hunk 1 above)
Comment (rcchen) [inline src/components/componentContainer.ts]: It's still there, https://github.com/palantir/plottable/pull/1958/files#diff-039fe0a9f2bd34b0a057ca4ed08395ddR31

File: src/components/component.ts [hunk 2]
```diff
@@ -500,7 +500,7 @@ module Plottable {
      * Removes a Component from the DOM and disconnects it from everything it's
      * listening to (effectively destroying it).
      */
-    public remove() {
+    public destroy() {
       this._removed = true;
```
Comment (jtlan) [inline src/components/component.ts]: Probably want to rename this var to `_destroyed` for consistency.

**Comment by jtlan** [comment]
[minor] rename internal variable to match up with method name.

**Pull request merged**